### PR TITLE
Don't consider scheduler idle while executing `Scheduler.update_graph`

### PR DESCRIPTION
**Pull request by hendrikmakait**
Partially addresses #8876 

- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

**Review by jrbourbeau — COMMENTED** [review]
Nice, thanks @hendrikmakait. I think this will help when dealing with large graphs

File: distributed/scheduler.py [hunk 1]
```diff
@@ -4913,8 +4916,11 @@ async def update_graph(
                     # (which may not have been added to who_wants yet)
                     client=client,
                 )
-        end = time()
-        self.digest_metric("update-graph-duration", end - start)
+        finally:
```
Comment (jrbourbeau) [inline distributed/scheduler.py:4919]: Just noting that this `finally` block is being used to ensure we _always_ decrement the `_active_graph_updates` counter

File: distributed/tests/test_scheduler.py [hunk 2]
```diff
@@ -2389,6 +2389,33 @@ async def test_idle_timeout(c, s, a, b):
...
+    assert s.idle_since < beginning
+    await c.submit(lambda x: x, 1)
+    # The cluster may be considered not idle because of the unit of work
+    s.check_idle()
+    # Now the cluster must be idle
+    assert s.check_idle() is not None
+    end = time()
+    assert beginning <= s.idle_since
+    assert s.idle_since <= end
+    assert plugin.idle_during_update_graph is None
```
Comment (jrbourbeau) [inline distributed/tests/test_scheduler.py:2416]: [Non-blocking nit] It took me a couple of minutes to realize that this was the particularly meaningful check. Might be worth adding a comment or something to highlight that. 

```suggestion
    # Ensure the cluster isn't idle while `Scheduler.update_graph` was being run
    assert plugin.idle_during_update_graph is None
```

File: distributed/tests/test_scheduler.py [hunk 3]
```diff
@@ -2389,6 +2389,33 @@ async def test_idle_timeout(c, s, a, b):
     pc.stop()
 
 
+@gen_cluster(client=True)
+async def test_idle_during_update_graph(c, s, a, b):
+    class UpdateGraphTrackerPlugin(SchedulerPlugin):
+        def start(self, scheduler):
+            self.scheduler = scheduler
+            self.idle_during_update_graph = None
```
Comment (jrbourbeau) [inline distributed/tests/test_scheduler.py:2397]: [Non-blocking nit] This is totally fine as is but maybe don't even assign `self.idle_during_update_graph` until inside the `update_graph` plugin hook? `assert plugin.idle_during_update_graph is None` _could_ mean that the `update_graph` hook didn't run at all. 

```suggestion
```

**Review by jrbourbeau — APPROVED** [review]
Thanks @hendrikmakait, this LGTM. I left a couple of non-blocking comments. Feel free to include or ignore.

File: distributed/tests/test_scheduler.py (hunk 3 above)
Comment (hendrikmakait) [reply distributed/tests/test_scheduler.py:2397]: I've slightly adjusted this such that `idle_during_update_graph` is initialized to `None` and set to `True` or `False` during the hook.

File: distributed/tests/test_scheduler.py [hunk 4]
```diff
@@ -2389,6 +2389,34 @@ async def test_idle_timeout(c, s, a, b):
...
+
+        def update_graph(self, *args, **kwargs):
+            self.idle_during_update_graph = self.scheduler.check_idle() is None
```
Comment (jrbourbeau) [inline distributed/tests/test_scheduler.py:2400]: The idle check should be "is not `None`" here I think 

```suggestion
            self.idle_during_update_graph = self.scheduler.check_idle() is not None
```

Comment (hendrikmakait) [reply distributed/tests/test_scheduler.py:2400]: 🤦

**Pull request merged**